code/go/0chain.net/blobber/zcn.go

```diff
@@ -26,33 +26,33 @@ func registerOnChain() error {
 		return nil
 	}
 
-	if err := handler.WalletRegister(); err != nil {
-		fmt.Println(err.Error() + "\n")
-		panic(err)
+	var err error
+
+	err = handler.WalletRegister()
+	if err != nil {
+		return err
 	}
 	fmt.Print("	[OK]\n")
 
-	var err error
+	err = filestore.GetFileStore().CalculateCurrentDiskCapacity()
+	if err != nil {
+		return err
+	}
+
 	// setup blobber (add or update) on the blockchain (multiple attempts)
 	for i := 1; i <= 10; i++ {
 		if i == 1 {
 			fmt.Printf("\r	+ connect to sharders:")
 		} else {
-
-			for n := ATTEMPT_DELAY; n < 1; n-- {
+			for n := ATTEMPT_DELAY; n > 0; n-- {
 				if n == 1 {
-					fmt.Printf("\r	+ [%v/10]connect to sharders :", i)
+					fmt.Printf("\r	+ [%v/10]connect to sharders:      ", i)
 				} else {
-					fmt.Printf("\r	+ [%v/10]connect to sharders %v:", i, n)
+					fmt.Printf("\r	+ [%v/10]connect to sharders: %.2vs", i, n)
 				}
-				time.Sleep(1 * time.Second)
-			}
 
-		}
-
-		err = filestore.GetFileStore().CalculateCurrentDiskCapacity()
-		if err != nil {
-			continue
+				<-time.After(1 * time.Second)
+			}
 		}
 
 		err = handler.RegisterBlobber(common.GetRootContext())
```

Comment:
this can be changed to 

```go
if err == nil {
    break;
}
```

or even can add to loop checks, 
```go
	for i := 1; i <= 10 && err != nil; i++ {
```

Reply:
fixed

Reply:
fixed